### PR TITLE
Remove deprecated `extensions` API group in document

### PR DESCRIPTION
**Pull request by Sea-n**
The `extensions` API group is deprecated.

### Changes in ABAC page
- Use `networking.k8s.io` instead of `extensions` in `apiGroups` field.
    - Ref: kubernetes/kubernetes#74057
- Add more examples.

### Changes in RBAC page
- Remove deprecated `extensions` API group for Jobs
    - Ref: kubernetes/kubernetes#36355
- Remove deprecated `extensions` API group for Deployments
    - Ref: kubernetes/kubernetes#70672

### CODE REVIEW

File: content/en/docs/reference/access-authn-authz/rbac.md [hunk 1]
```diff
@@ -407,7 +407,7 @@ rules:
   # objects is "pods"
   resources: ["pods"]
   verbs: ["get", "list", "watch"]
-- apiGroups: ["batch", "extensions"]
+- apiGroups: ["batch", "networking.k8s.io"]
```
Comment (liggitt) [inline content/en/docs/reference/access-authn-authz/rbac.md:410]: jobs don't exist in the networking.k8s.io group... if you want to use a multi-group example, `"events"` resources in the core `""` and `"events.k8s.io"` API groups would be a current example

Comment (Sea-n) [reply content/en/docs/reference/access-authn-authz/rbac.md]: Oh! Thanks very much for pointing it out and providing a great example!

I have pushed a new commit, could you please check the correctness again? :bowing_man:

File: content/en/docs/reference/access-authn-authz/rbac.md [hunk 2]
```diff
@@ -383,21 +383,21 @@ rules:
   verbs: ["get", "list", "watch"]
 ```
 
-Allow reading/writing Deployments (at the HTTP level: objects with `"deployments"`
-in the resource part of their URL) in both the `"extensions"` and `"apps"` API groups:
+Allow reading/writing Events (at the HTTP level: objects with `"events"` in the
+resource part of their URL) in both the core `""` and `"events.k8s.io"` API groups:
```
Comment (sftim) [inline content/en/docs/reference/access-authn-authz/rbac.md:387]: I'd use Deployment, in its GA API group (`apps`). People rarely patch Events.

File: content/en/docs/reference/access-authn-authz/abac.md [hunk 3]
```diff
@@ -33,13 +33,13 @@ properties:
...
       - `apiGroup`, type string; an API group.
-        - Ex: `extensions`
+        - Ex: `apps`, `networking.k8s.io`
         - Wildcard: `*` matches all API groups.
       - `namespace`, type string; a namespace.
         - Ex: `kube-system`
         - Wildcard: `*` matches all resource requests.
       - `resource`, type string; a resource type
-        - Ex: `pods`
+        - Ex: `pods`, `events`
```
Comment (sftim) [inline content/en/docs/reference/access-authn-authz/abac.md:42]: Why this change?

**Review by sftim — COMMENTED** [review]
Thanks for proposing the update.

One question, one recommended tweak.

File: content/en/docs/reference/access-authn-authz/rbac.md (hunk 2 above)
Comment (Sea-n) [reply content/en/docs/reference/access-authn-authz/rbac.md:387]: Thanks for letting me know that `events` is not a common resource. :sweat_smile:

I think this section ([Role examples](https://kubernetes.io/docs/reference/access-authn-authz/rbac/#role-examples)) is about how to write rules instead of most used resources:

1. Reading `pods` in the core API group
    - The most simple one
2. R/W `deployments` in `extensions` and `apps` API group
    - Multiple API groups
3. Reading `pods` in core API group and R/W `jobs` in `batch` and `extensions`
    - Multiple rules
4. Allow reading a ConfigMap named `my-config`
    - Must be bound with a RoleBinding to limit to a single ConfigMap in a single namespace.
5. Allow reading the resource "nodes" in the core group
    - Because a Node is cluster-scoped, this must be in a ClusterRole bound with a ClusterRoleBinding to be effective.
6. Allow GET and POST requests to the non-resource endpoint
    - Example for `nonResourceURLs`

Since `extensions` is deprecated, I think it's good to replace it with another (although not that common) example for multiple API groups.

There are only three resources that appeared in multiple API groups:
```
NAME      APIVERSION                NAMESPACED    KIND
events    v1                        true          Event
events    events.k8s.io/v1          true          Event
nodes     v1                        false         Node
nodes     metrics.k8s.io/v1beta1    false         NodeMetrics
pods      v1                        true          Pod
pods      metrics.k8s.io/v1beta1    true          PodMetrics
```

I am still a beginner at Kubernetes, could you give me some advice about which could be a better example for this page?

File: content/en/docs/reference/access-authn-authz/abac.md (hunk 3 above)
Comment (Sea-n) [reply content/en/docs/reference/access-authn-authz/abac.md:42]: I think it would be better to provide multiple examples. Just like you said that it's not common to see `events`, could you give me some advice on this?

File: content/en/docs/reference/access-authn-authz/rbac.md (hunk 2 above)
Comment (liggitt) [reply content/en/docs/reference/access-authn-authz/rbac.md:387]: The `nodes` and `pods` in v1 and metrics API groups are not the same resource, so I wouldn't expect to grant access to both in the same rule.

`events` are the only remaining built-in resource that appears in multiple API groups (we had a lot in the `extensions` API group but those got migrated to dedicated groups and the `extensions` API group was deprecated and disabled)

Comment (Sea-n) [reply content/en/docs/reference/access-authn-authz/rbac.md:387]: Thanks for clarifying! Now I think it's better to change `events` example back to `deployments`. :blush:

File: content/en/docs/reference/access-authn-authz/abac.md (hunk 3 above)
Comment (Sea-n) [reply content/en/docs/reference/access-authn-authz/abac.md]: I've change it to `deployments`, hope it will be more useful for users.

**Comment by sftim** [comment]
@kubernetes/sig-auth-pr-reviews does this look right?

File: content/en/docs/reference/access-authn-authz/rbac.md [hunk 4]
```diff
@@ -384,20 +384,20 @@ rules:
...
-in the resource part of their URL) in both the `"extensions"` and `"apps"` API groups:
+in the resource part of their URL) in the `"apps"` API groups:
 
 ```yaml
 rules:
-- apiGroups: ["extensions", "apps"]
+- apiGroups: ["apps"]
   #
-  # at the HTTP level, the name of the resource for accessing Deployment
+  # at the HTTP level, the name of the resource for accessing Deployments
```
Comment (sftim) [inline content/en/docs/reference/access-authn-authz/rbac.md:393]: nit:
```suggestion
  # at the HTTP level, the name of the resource for accessing Deployment
```
In the URL it's `/deployments/`; in text, it's “Deployments” or “Deployment objects”.

**Review by sftim — COMMENTED** [review]
Markdown LGTM

File: content/en/docs/reference/access-authn-authz/rbac.md (hunk 4 above)
Comment (Sea-n) [reply content/en/docs/reference/access-authn-authz/rbac.md]: Oh no, it's a typo when I manually revert the change. :cry:

How could I miss that, thanks for pointing it out! :bowing_man:

**Comment by Sea-n** [comment]
Hi @liggitt,

I've force-pushed the changes according to your suggestions, could you take a look? :bowing_man:
